frontend/src/TimePreference.js

```diff
@@ -18,20 +21,29 @@ class TimePreference extends React.Component {
     this.props.updateTimeEndPreference(this.props.id, event.target.value);
   }
 
+  handleDeleteTimePreference() {
+    this.props.deleteTimePreference(this.props.id);
+  }
+
   render() {
     return (
-      <CardContent>
-        <TextField
-          label="Start time: " type="time"
-          defaultValue="08:00" onChange={this.handleStartTimeChange}
-          InputLabelProps={{ shrink: true, }} inputProps={{ step: 300, }}
-        />
-        <TextField
-          label="End time: " type="time"
-          defaultValue="17:00" onChange={this.handleEndTimeChange}
-          InputLabelProps={{ shrink: true, }} inputProps={{ step: 300, }}
-        />
-      </CardContent>
+      <Card>
+        <CardContent>
+          <TextField
+            label="Start time: " type="time"
+            defaultValue="08:00" onChange={this.handleStartTimeChange}
```

Comment:
One thing I've noticed (in very limited testing) is that this default value isn't applied, so if the user submits without actually changing the value it submits an empty string in the Start Time (or End Time) fields.
I don't think there's much one can do to change this aside from either removing the "defaultValue" (so the user knows that they need to change the input) or doing something fancier with state or something like that.
This isn't a big deal but I thought I'd mention it.

Reply:
Yes - thank you for bringing it up, I was going to mention that as well because I noticed it (and planned on changing it) too. I didn't want the user to think they didn't have to put in a time if it was the same as the default, so I was just going to remove it. I'll do that now since we've brought it up!